### DISCUSSION TITLE
build of libsndfile does not find codec libraries

### DISCUSSION THREAD

**Discussion by nwh** [Homebrew/discussions#209] [Everyday usage] [answered]
Hello! I am trying to get support for FLAC via `libsndfile`.  This stopped working with brewed libsndfile 1.0.30.  `brew reinstall -s libsndfile` was working for a while, but now after updating command line tools the libsndfile build fails to see the codec libraries.

```
$ brew reinstall -s -v libsndfile
# ...
checking for pkg-config... /usr/local/Homebrew/Library/Homebrew/shims/mac/super/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for flac >= 1.3.1 ... link failed ... no
checking for ogg >= 1.3.0 ... link failed ... no
checking for vorbis >= 1.2.3 ... link failed ... no
checking for vorbisenc >= 1.2.3 ... link failed ... no
checking for opus >= 1.1 ... link failed ... no

configure: WARNING: *** One or more of the external libraries (ie libflac, libogg and
configure: WARNING: *** libvorbis) is either missing (possibly only the development
configure: WARNING: *** headers) or is of an unsupported version.
configure: WARNING: ***
configure: WARNING: *** Unfortunately, for ease of maintenance, the external libs
configure: WARNING: *** are an all or nothing affair.
```

Any thoughts?

**Comment by nwh** [comment, marked as answer]
This appears to have been fixed upstream: <https://github.com/libsndfile/libsndfile/issues/642>

**Reply by nwh** [reply]
Sounds like this fix will be in the next release.

**Reply by carlocab** [reply]
You may want to submit a PR for this.

There are many examples of formulae with upstream patches: just `grep 'patch do'` in `$(brew --repo homebrew/core)`. I think `libfreefanfare` serves as a good example, as the preceding comments are important. (You can do `brew cat libfreefanfare` to see.)

I think the patch you need is in this commit: https://github.com/libsndfile/libsndfile/pull/644/commits/ecd63961da5f8c572645aae476f19a1343e755b9

To get the plaintext patch for the formula, append `.patch?full_index=1` to the URL. (Hopefully, the patch applies cleanly.)

**Reply by nwh** [reply]
Thanks for the suggestion. I will give it a shot!

**Reply by carlocab** [reply]
Glad to hear that. Good luck!

Feel free to ask (or tag me in your PR) if you have any problems, and I'll do my best to help.

**Reply by nwh** [reply]
Should I manually set `revision 1`?

**Reply by carlocab** [reply]
This patch is meant to fix something that used to work, right? If so, then yes. If not, you may still want to set it if it's an important feature of the package.

**Reply by nwh** [reply]
I am seeing `brew audit --strict libsndfile` errors, but these appear to be from failures in installing the Ruby packages required to run audit.

**Reply by carlocab** [reply]
Might be worth opening the PR anyway, as long as you're able to build the formula from source `brew install -s libsndfile` and can confirm that you have your codec libraries.
